python/ray/tune/tests/test_result_grid.py

```diff
@@ -123,87 +81,6 @@ def f(config):
     assert result.checkpoint is None
 
 
-@pytest.mark.parametrize("to_object", [False, True])
-def test_result_grid_future_checkpoint(ray_start_2_cpus, to_object):
-    trainable_cls = get_trainable_cls("__fake")
-    trial = Trial("__fake", stub=True)
-    trial.config = {"some_config": 1}
-    trial.run_metadata.last_result = {"some_result": 2, "config": trial.config}
-
-    trainable = ray.remote(trainable_cls).remote()
-    ray.get(trainable.set_info.remote({"info": 4}))
-
-    if to_object:
-        checkpoint_data = trainable.save_to_object.remote()
-    else:
-        checkpoint_data = trainable.save.remote()
-
-    trial.on_checkpoint(
-        _TrackedCheckpoint(checkpoint_data, storage_mode=CheckpointStorage.MEMORY)
-    )
-    trial.run_metadata.pickled_error_filename = None
-    trial.run_metadata.error_filename = None
-
-    class MockExperimentAnalysis:
-        trials = []
-        trial_dataframes = None
-
-    result_grid = ResultGrid(MockExperimentAnalysis())
-
-    # Internal result grid conversion
-    result = result_grid._trial_to_result(trial)
-    assert isinstance(result.checkpoint, Checkpoint)
-    assert isinstance(result.metrics, dict)
-    assert isinstance(result.config, dict)
-    assert result.metrics_dataframe is None
-    assert result.config == {"some_config": 1}
-    assert result.metrics["config"] == result.config
-
-    # Load checkpoint data (see ray.rllib.algorithms.mock.MockTrainer definition)
-    with result.checkpoint.as_directory() as checkpoint_dir:
-        with open(os.path.join(checkpoint_dir, "mock_agent.pkl"), "rb") as f:
-            info = pickle.load(f)
-            assert info["info"] == 4
-
-
-def test_best_result(ray_start_2_cpus):
```

Comment:
This PR removes quite a few tests - is the functionality covered somewhere else? Do we have to migrate this later?

Reply:
Most of these have been consolidated in the new tests. I've also removed some that were just testing behavior that was passing through from `ExperimentAnalysis`.